utils/land-patterns/pose-arrangement.stanza

```diff
@@ -0,0 +1,127 @@
+defpackage ocdb/land-patterns/pose-arrangement:
+  import core
+  import collections
+  import math
+  import jitx
+  
+defstruct Affine:
+  matrix: [
+    [Double, Double, Double],
+    [Double, Double, Double],
+    [Double, Double, Double]
+  ]
+
+defn Affine () -> Affine:
+  Affine([
+    [1.0, 0.0, 0.0]
+    [0.0, 1.0, 0.0]
+    [0.0, 0.0, 1.0]
+  ])
+
+defn zero () -> Affine:
+    Affine([
+    [0.0, 0.0, 0.0]
+    [0.0, 0.0, 0.0]
+    [0.0, 0.0, 0.0]
+  ])
+
+defn times (a:Affine, b:Affine) -> Affine:
+  val [A, B] = [matrix(a), matrix(b)]
+  val C = Array<Array<Double>>(3)
+  
+  for i in 0 to 3 do:
+    C[i] = Array<Double>(3)
+    for j in 0 to 3 do: 
+      C[i][j] = 0.0
+      for k in 0 to 3 do: 
+        C[i][j] = C[i][j] + A[i][k] * B[k][j]
+  Affine([
+    [C[0][0], C[0][1], C[0][2]]
+    [C[1][0], C[1][1], C[1][2]]
+    [C[2][0], C[2][1], C[2][2]]
+  ])
+
+defn scale (x:Double, y:Double) -> Affine:
+  Affine([
+    [x, 0.0, 0.0]
+    [0.0, y, 0.0]
+    [0.0, 0.0, 1.0]
+  ])
+
+defn flip-x () -> Affine:
+  scale(-1.0, 1.0)
+
+defn flip-y () -> Affine:
+  scale(1.0, -1.0)
+
+defn translate (x:Double, y:Double) -> Affine:
+  Affine([
+    [1.0, 0.0, x]
+    [0.0, 1.0, y]
+    [0.0, 0.0, 1.0]
+  ])
+ 
+defn rotation (degrees:Double) -> Affine:
+  val theta = PI * degrees / 180.0
+  Affine([
+    [cos(theta), -1.0 * sin(theta), 0.0],
+    [sin(theta), cos(theta), 0.0],
+    [0.0, 0.0, 1.0]
+  ])
+
+defn times (p:Point, a:Affine) -> Point:
+  val p1 = Array<Double>(3)
```

Comment:
is there a way to do something like `val p1 = Array<Double>(3) [x(p) y(p) 1.0]`
if not that would be great to have it @CuppoJava